packages/core/src/agent/AgentMessage.ts

```diff
@@ -1,36 +1,24 @@
-import type { ParsedMessageType } from '../utils/messageType'
-import type { Constructor } from '../utils/mixins'
+import type { ServiceDecorator } from '../decorators/service/ServiceDecorator'
+import type { ReturnRouteTypes } from '../decorators/transport/TransportDecorator'
+import type { DIDCommMessageVersion } from './didcomm/types'
 
-import { AckDecorated } from '../decorators/ack/AckDecoratorExtension'
-import { AttachmentDecorated } from '../decorators/attachment/AttachmentExtension'
-import { L10nDecorated } from '../decorators/l10n/L10nDecoratorExtension'
-import { ServiceDecorated } from '../decorators/service/ServiceDecoratorExtension'
-import { ThreadDecorated } from '../decorators/thread/ThreadDecoratorExtension'
-import { TimingDecorated } from '../decorators/timing/TimingDecoratorExtension'
-import { TransportDecorated } from '../decorators/transport/TransportDecoratorExtension'
-import { JsonTransformer } from '../utils/JsonTransformer'
-import { replaceNewDidCommPrefixWithLegacyDidSovOnMessage } from '../utils/messageType'
+export interface AgentMessage {
+  readonly type: string
 
-import { BaseMessage } from './BaseMessage'
+  get version(): DIDCommMessageVersion
+  get id(): string
+  get sender(): string | undefined
```

Comment:
is sender always a did? what are the values/types it can have? Would be good to note this either in the type or the property name